File: Samples/ObjCSample/ObjCSample/NativeAdInTableViewController/ListItem.m
//
//  AdListItem.m
//  ObjCSample
//
//  Created by Julien Gomez on 11/09/2015.
//  Copyright © 2015 Smart AdServer. All rights reserved.
//

#import "ListItem.h"

@implementation ListItem

- (NSString *)title {
    if (_isAd) {
        return [_nativeAd title];
    } else {
        return [_rssElement title];
    }
}


- (NSString *)subtitle {
    if (_isAd) {
        return [_nativeAd subtitle];
    } else {
        return [_rssElement postDescription];
    }
}


- (NSURL *)link {
    if (_isAd) {
        return nil;
    } else {
        return [_rssElement link];
    }
}


- (NSURL *)imageURL {
    if (_isAd) {
        return [[_nativeAd icon] URL];
    } else {
        return [_rssElement imageURL];
    }
}

- (NSString *)publishedDate {
    if (_isAd) {
        return nil;
    } else {
        return [_rssElement publishedDate];
    }
}

@end

File: Samples/SwiftSample/SwiftSample/NativeAdInTableViewcontroller/ListItem.swift
//
//  ListItem.swift
//  SwiftSample
//
//  Created by Loïc GIRON DIT METAZ on 07/09/2015.
//  Copyright © 2015 Smart AdServer. All rights reserved.
//

import UIKit


// Represents an item in the table view of the NativeAdInTableViewController
protocol ListItem {
    
    func title() -> String
    func subtitle() -> String
    func image() -> String
    
}
